go/pkg/common/utils.go

```diff
@@ -124,3 +125,32 @@ func SyncMapLength(m *sync.Map) int {
 	})
 	return i
 }
+
+// RetryUntilSuccessful calls callback every interval until duration until it returns false (to not retry)
+func RetryUntilSuccessful(duration time.Duration,
+	interval time.Duration,
+	callback func() (bool, error)) error {
```

Comment:
```suggestion
func RetryUntilSuccessful(timeout time.Duration,
	backoff time.Duration,
	callback func() (bool, error)) error {
```

Reply:
Also maybe add a `RetryUntilSuccessfulWithResult` like in nuclio/provazio and then `RetryUntilSuccessful` wraps it and just doesn't return the result

Reply:
You mean return the error?
I do want the actual error to be returned in case all retries failed, so I can handle it myself.

Reply:
No. I mean the return value of the callback if it has one